### PR TITLE
Add Capability to enable check via Visual Studio SDK

### LINKED ISSUE #341: Feature Request: Automatically add a MSBuild property (IsDatabaseProject) via the SDK
I have run into a scenario where it would be very handy to know if a given project is a "Database Project" or not. Similar to what the Test SDK does with adding the "_IsTestProject_" property it would be nice if this project type SDK could do the same and add one named "_IsDatabaseProject_".

I've looked at what is currently present for an interim solution for anything I could inspect and use to evaluate, but I'd much rather have a purpose-built approach to leverage.

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #341

### CODE REVIEW

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets [hunk 1]
```diff
@@ -23,6 +23,9 @@
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" Condition=" '$(CommonTargetsPath)' == '' " />
 
   <ItemGroup>
+    <!-- Enable detection of this project SDK - see https://github.com/microsoft/VSProjectSystem/blob/master/doc/overview/about_project_capabilities.md -->
+    <ProjectCapability Include="BuildTSqlScript" />
```
Comment (jmezach) [inline src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:27]: Is this a known capability? Or is this just something you've made up?

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:27]: There is no suitable existing, so I made this up based on [the guidance here](https://github.com/microsoft/VSProjectSystem/blob/master/doc/overview/about_project_capabilities.md)

Comment (jmezach) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:27]: Shouldn't we open an issue on that repo then as well to discuss that capability?

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:27]: https://github.com/microsoft/VSProjectSystem/issues/364

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: I updated the name based on a re-read of the docs

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: With the prefix I think we do not need MS "approval"

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: Given the [discussion here](https://github.com/microsoft/VSProjectSystem/issues/364#issuecomment-1472931004) I believe we are good to go @jmezach

**Comment by ErikEJ** [comment]
@jeffrosenberg Are you available to have a look at this?

**Review by jeffrosenberg — APPROVED** [review]
My apologies, I was away on vacation. LGTM!

**Comment by ErikEJ** [comment]
@jeffrosenberg No need to apologize! 😄

**Comment by jmezach** [comment]
LGTM :).

**Review by jmezach — APPROVED** [review]

**Pull request merged**